From ec46246baa96ad5ce0b6eb9a36dd76532133b0c2 Mon Sep 17 00:00:00 2001
From: Misael Lopez Cruz <misael.lopez@ti.com>
Date: Tue, 14 Dec 2010 18:29:02 -0600
Subject: OMAP4: McBSP: Add wakeup sources for threshold mode

McBSP driver uses smart-idle and wakeup feature for threshold mode,
but wakeup sources are not enabled. Add XRDYEN and RRDYEN wakeup
sources for threshold mode. Wakeup sources are not required for
element mode because idle mode is set to no-idle.

Change-Id: Ic94ad8b25fac014142a768f70eced91e3cdcf8ae
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
---
 arch/arm/plat-omap/mcbsp.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/arm/plat-omap/mcbsp.c b/arch/arm/plat-omap/mcbsp.c
index aa0f4b7..4a3f4f2 100644
--- a/arch/arm/plat-omap/mcbsp.c
+++ b/arch/arm/plat-omap/mcbsp.c
@@ -682,6 +682,7 @@ static inline void omap34xx_mcbsp_request(struct omap_mcbsp *mcbsp)
 	if (cpu_is_omap34xx() || cpu_is_omap44xx()) {
 
 		if (mcbsp->dma_op_mode == MCBSP_DMA_MODE_THRESHOLD) {
+			MCBSP_WRITE(mcbsp, WAKEUPEN, XRDYEN | RRDYEN);
 			omap_hwmod_enable_wakeup(mcbsp->oh[0]);
 			omap_hwmod_set_slave_idlemode(mcbsp->oh[0],
 						HWMOD_IDLEMODE_SMART);
-- 
1.6.0.4

